pandas/tests/reshape/test_concat.py

```diff
@@ -177,7 +177,15 @@ def test_concatlike_same_dtypes(self):
             tm.assert_series_equal(res, exp, check_index_type=True)
 
             # cannot append non-index
-            msg = "cannot concatenate a non-NDFrame object"
+
+            # multiline string concat is safer with \n and whitespace
+            # for regex (see concat.py:266
+
+            # (.+?) may be too general but we are more concerned
+            # with the message as a whole, anyways
+            msg = """cannot concatenate object of type \"(.+?)\";"""
+            msg += " only pd.Series, pd.DataFrame, and pd.Panel"
+            msg += " objs are valid"
```

Comment:
same comment here as above. 
But also, I think it is fine to only match a part of the error message if that makes it easier.

Reply:
Thanks, I'll clean this up (which will eliminate the confusing comment as well).